gnovm/pkg/gnolang/preprocess.go

```diff
@@ -81,15 +81,40 @@ func PredefineFileSet(store Store, pn *PackageNode, fset *FileSet) {
 	// Finally, predefine other decls and
 	// preprocess ValueDecls..
 	for _, fn := range fset.Files {
-		for i := 0; i < len(fn.Decls); i++ {
+		lenDecls := len(fn.Decls)
+		for i := 0; i < lenDecls; i++ {
 			d := fn.Decls[i]
 			if d.GetAttribute(ATTR_PREDEFINED) == true {
 				// skip declarations already predefined (e.g.
 				// through recursion for a dependent)
 			} else {
```

Comment:
Nitpick: you can simplify this conditional as:
```go
if d.GetAttribute(ATTR_PREDEFINED) {
    // skip declarations already predefined (e.g.
	// through recursion for a dependent)
    continue
}

...
```

This way there is no `else` block. I know this code is leftover from the previous implementation, but I think this really increases the readability 🙏

Reply:
The way you wrote the condition, it is not possible.
That function returns `interface{}` so you can't directly use that as a boolean value.